x/metadata/client/cli/cli_test.go

```diff
@@ -1410,6 +1410,7 @@ func runTxCmdTestCases(s *IntegrationCLITestSuite, testCases []txCmdTestCase) {
 				require.EqualError(t, err, tc.expectErrMsg, "%s expected error message", cmdName)
 			} else if tc.expectErr {
 				require.Error(t, err, "%s expected error", cmdName)
+				require.EqualError(t, err, tc.expectErrMsg)
```

Comment:
At this point, you know that `tc.expectErrMsg` is `""`.

Reply:
No, the error message is set in the test as a string.  I'm testing that the cli command validates the data correctly before broadcasting it to the message server.  I find this valuable to assure it fails were I expected it to fail.

Reply:
Four lines up, there is `if len(tc.expectErrMsg) > 0 {` which does the `EqualError` check. So if it gets to this `else if tc.expectErr`, you know that len(tc.expectErrMsg) is zero.

There's a good chance we could completely remove the `tc.expectErr` flag from these tests. I was going to, but since we were both in there making changes last week, I figured that'd be too much of a merge pain. It's only there when we want to make sure an error is returned, but we don't care what the error is.

Reply:
Also, there is no reason to ever call `Error` followed by `EqualError`.  The `EqualError` check includes making sure that the provided error isn't nil.

Reply:
Ah, I see what you mean with it being checked above.  Totally, missed that.  However, on the note about caring what the error message is, I agree that we really don't care what the error message itself is.  It is more for checking that it failed in the place I expected.  I find it valuable and have caught many issues when using it.

Reply:
99% of the test setups we have that include both an expected error message as well as a boolean to indicate whether or not we expect an error can be simplified by removing the boolean and using the existence of a non-empty error string to indicate that we expect the test to fail. That's my point about the `tc.expectErr` flag there. If it's always true when `tc.expectErrMsg` is not empty, then it's not needed. The only time we would want such a flag in the test case setup is when we want to run a test that we expect will give an error, but we don't want to define the expected error.

But if `tc.expectErrMsg` is not empty, then we want to make sure that a) an error was generated, and b, it equals the provided string. Both of these are done with `EqualError`. If `tc.expectErrMsg` is an empty string, but `tc.expectErr` is true, then we expect an error, but don't care about the message. So we should use `Error` and we should NOT call `EqualError` because we don't have the expected error message to compare against.

In fact, if your tests are passing with this line there, it shows that there isn't a single test case that has `len(tc.expectErrMsg) == 0 && tc.expectErr == true`. Otherwise the `EqualError` test would fail because you're comparing the error's message to an empty string, and it's highly unlikely that the error's message is an empty string.

Reply:
Sorry, totally misunderstood what you were trying to say in your comment (thought you were saying we shouldn't check what the error message is).  Yes, this made sense when I took a deeper look at where my mistake was.